server/src/settings.ts

```diff
@@ -1,17 +1,53 @@
 import type { Configuration } from 'vscode-languageserver/lib/common/configuration';
+import { URI } from 'vscode-uri';
 
 import type { FluenceCli } from './cli';
 
+type Imports = string[];
+
 export interface Settings {
-    imports: string[];
+    imports: Imports;
     enableLegacyAutoImportSearch: boolean;
 }
 
-function addImports(settings: Settings, imports?: string[]): Settings {
-    return {
-        ...settings,
-        imports: [...settings.imports, ...(imports ?? [])],
-    };
+/**
+ * Document info stored in `SettingsManager` cache
+ */
+class DocumentInfo {
+    /* Settings from configuration (or default) */
+    private settings: Settings;
+    /* Additional imports from CLI */
+    private imports: Imports = [];
+    private importsLastUpdated = 0;
+    private importsUpdateRequested = true;
+
+    constructor(settings: Settings) {
+        this.settings = settings;
+    }
+
+    getSettings(): Settings {
+        return {
+            ...this.settings,
+            imports: [...this.settings.imports, ...(this.imports ?? [])],
```

Comment:
```suggestion
            imports: [...this.settings.imports, ...this.imports],
```
`this.imports` always has an array type, right?